vim/core/oni-plugin-buffers/index.js

```diff
@@ -0,0 +1,111 @@
+// @ts-check
+const path = require("path")
+
+const activate = Oni => {
+    const menu = Oni.menu.create()
+
+    const truncateFilePath = filepath => {
+        const sections = filepath.split(path.sep)
+        const folderAndFiles = sections.slice(-2)
+        return folderAndFiles.join(path.sep)
+    }
+
+    const updateBufferList = (Oni, menu) => {
+        const buffers = Oni.editors.activeEditor.getBuffers()
+        const active = Oni.editors.activeEditor.activeBuffer.filePath
+
+        const bufferMenuItems = buffers.map(b => ({
+            label: `${active === b.filePath ? b.id + " %" : b.id}`,
+            detail: truncateFilePath(b.filePath),
+            icon: Oni.ui.getIconClassForFile(b.filePath),
+            pinned: active === b.filePath,
+            metadata: { filePath: b.filePath, id: b.id },
+        }))
+
+        return bufferMenuItems
+    }
+
+    const createBufferList = () => {
+        const buffers = updateBufferList(Oni, menu)
+        menu.show()
+        menu.setItems(buffers)
+    }
+
+    const toggleBufferList = () => {
+        !menu.isOpen() ? createBufferList() : menu.hide()
+    }
+
+    const deleteBuffer = async menu => {
+        if (menu.selectedItem) {
+            await Oni.editors.activeEditor.bufferDelete(menu.selectedItem.metadata.id)
+            // This line forces vim to navigate to the next buffer after deleting
+            // NOTE: This is essential as otherwise Oni does not properly register the buffer
+            // delete event
+            await Oni.editors.activeEditor.neovim.command("bn")
+            menu.hide()
+        }
+    }
+
+    const openBuffer = (menu, orientation) => {
+        if (menu.selectedItem && menu.isOpen()) {
+            const buffers = Oni.editors.activeEditor.getBuffers()
+            try {
+                const { filePath = "" } = menu.selectedItem.metadata
+                Oni.editors.activeEditor.openFile(filePath, orientation)
+            } catch (e) {
+                console.warn("[Oni Buffer Plugin Error]: ", e)
+            } finally {
+                menu.hide()
+            }
+        }
+        return
+    }
+
+    Oni.commands.registerCommand({
+        command: "bufferlist.delete",
+        name: "Delete Selected Buffer",
```

Comment:
Could we add a common 'prefix' for these, like:
````
name: "Buffer: Delete",
detail: "Delete selected buffer",
```

Trying to organize the commands a bit 😄

Reply:
@bryphe did you intend for the commands to be in caps with the space there, I've changed the command to be `buffer:action` as from looking at the commands none start with caps and none have spaces in them ❓

Reply:
Yeah, no set standard yet, but here's what I'm thinking:

- command: camel case / lowercase - ie, `workspace.openFolder`
- name: noun + verb, capitalized - `Workspace: Open Folder`
- detail: sentence case - `Set a folder as the current workspace` 
(not a great example, but hopefully that helps).

I think the commands you have look good - I was thinking for the `name`/`detail` fields.

Like looking at the command palette we have today:
![image](https://user-images.githubusercontent.com/13532591/35464236-c5f81064-02a9-11e8-95bb-0af13ef5eefb.png)

The 'Workspace' ones look good to me, but the 'Show Errors' looks like it's just kind of floating. It'd be nice to group it with some other error/diagnostic related commands (maybe like `Errors: Show`, `Errors: Go to next`, `Errors: Go to previous`, etc..)